data_dictionary/data_dictionary.cc

```diff
@@ -362,34 +362,15 @@ std::ostream& keyspace_metadata::describe(std::ostream& os) const {
     return os;
 }
 
-std::ostream& operator<<(std::ostream& os, const keyspace_metadata& m) {
-    os << "KSMetaData{";
-    os << "name=" << m._name;
-    os << ", strategyClass=" << m._strategy_name;
-    os << ", strategyOptions={";
-    int n = 0;
-    for (auto& p : m._strategy_options) {
-        if (n++ != 0) {
-            os << ", ";
-        }
-        os << p.first << "=" << p.second;
-    }
-    os << "}";
-    os << ", cfMetaData={";
-    n = 0;
-    for (auto& p : m._cf_meta_data) {
-        if (n++ != 0) {
-            os << ", ";
-        }
-        os << p.first << "=" << p.second;
-    }
-    os << "}";
-    os << ", durable_writes=" << m._durable_writes;
-    os << ", userTypes=" << m._user_types;
-    os << "}";
-    return os;
 }
 
+auto fmt::formatter<data_dictionary::keyspace_metadata>::format(const data_dictionary::keyspace_metadata& m, fmt::format_context& ctx) const -> decltype(ctx.out()) {
+    return fmt::format_to(ctx.out(), "KSMetaData{{name={}, strategyClass={}, strategyOptions={{{}}}, cfMetaData={{{}}}, durable_writes={}, userTypes={}}}",
+            m.name(), m.strategy_name(), fmt::join(m.strategy_options(), ", "), fmt::join(m.cf_meta_data(), ", "), m.durable_writes(), m.user_types());
```

Comment:
Note: #include \<fmt/ranges.h> avoids the need for join()

Reply:
(but loses control over choice of separator and parentheses)

Reply:
Not this time, unfortunately:

```
/usr/include/fmt/ranges.h:555:30: error: no member named 'set_brackets' in 'fmt::detail::fallback_formatter<std::pair<const seastar::basic_sstring<char, unsigned int, 15>, seastar::basic_sstring<char, unsigned int, 15>>>'
    underlying_.underlying().set_brackets({}, {});
    ~~~~~~~~~~~~~~~~~~~~~~~~ ^
...
/usr/include/fmt/ranges.h:556:30: error: no member named 'set_separator' in 'fmt::detail::fallback_formatter<std::pair<const seastar::basic_sstring<char, unsigned int, 15>, seastar::basic_sstring<char, unsigned int, 15>>>'
    underlying_.underlying().set_separator(
    ~~~~~~~~~~~~~~~~~~~~~~~~ ^
```

Reply:
guess our own formatter for `std::pair<>` wins in the resolution game.